boa/src/builtins/object/tests.rs

```diff
@@ -289,3 +289,51 @@ fn object_is_prototype_of() {
 
     assert_eq!(context.eval(init).unwrap(), JsValue::new(true));
 }
+
+#[test]
+fn object_get_own_property_names_invalid_args() {
+    let mut context = Context::new();
+
+    let error_message = r#"Uncaught "TypeError": "cannot convert 'null' or 'undefined' to object""#;
+
+    assert_eq!(
+        forward(&mut context, r#"Object.getOwnPropertyNames()"#),
+        error_message
+    );
+    assert_eq!(
+        forward(&mut context, r#"Object.getOwnPropertyNames(null)"#),
+        error_message
+    );
+    assert_eq!(
+        forward(&mut context, r#"Object.getOwnPropertyNames(undefined)"#),
+        error_message
+    );
+}
+
+#[test]
+fn object_get_own_property_names() {
```

Comment:
We have a new API for tests that we haven't fully transitioned into. Here's an example on how to use it https://github.com/boa-dev/boa/blob/ffe20d8fffdd28f458f276cfa50817af4c80b015/boa/src/syntax/ast/node/iteration/tests.rs#L249-L271

The API should simplify a lot the nested `assert_eq!(forward(...))`

Reply:
Ah okay let me try them out